src/en/kami18/src/eu/kanade/tachiyomi/extension/en/kami18/Kami18.kt

```diff
@@ -0,0 +1,178 @@
+    }
+
+    override fun latestUpdatesRequest(page: Int): Request {
+        return GET("$baseUrl/albums?o=mr&page=$page", headers)
+    }
+
+    override fun latestUpdatesParse(response: Response) = popularMangaParse(response)
+
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request {
+        if (query.isNotEmpty()) {
+            val url = baseUrl.toHttpUrl().newBuilder().apply {
+                addPathSegment("search")
+                addPathSegment("photos")
+                addQueryParameter("main_tag", "5")
+                addQueryParameter("search_query", query)
+            }.build()
+            return GET(url, headers)
+        }
+        val url = baseUrl.toHttpUrl().newBuilder().apply {
+            var type = ""
+            var search = false
+            filters.forEach {
+                when (it) {
+                    is TypeFilter -> {
+                        type = it.getValue()
+                    }
+
+                    is SortFilter -> {
+                        addQueryParameter("o", it.getValue())
+                    }
+
+                    is TimelineFilter -> {
+                        addQueryParameter("t", it.getValue())
+                    }
+
+                    is TextFilter -> {
+                        if (it.state.isNotBlank()) {
+                            search = true
+                            addQueryParameter("search_query", it.state.replace(",", " "))
+                        }
+                    }
+                    else -> {}
+                }
+            }
+            addPathSegment(if (search) "search/photos" else "albums")
+            if (type.isNotEmpty()) addPathSegment(type)
+        }.build()
+
+        return GET(url, headers)
+    }
+
+    override fun searchMangaParse(response: Response) = popularMangaParse(response)
+
+    override fun getFilterList() = getFilters()
+
+    override fun mangaDetailsParse(response: Response) = SManga.create().apply {
+        val document = response.asJsoup()
+
+        description = buildString {
+            val desc = document.selectFirst("div[class*=p-t-5]:contains(description：)")?.ownText()?.substringAfter("：") ?: ""
+            append(desc)
+            append("\n\n", document.select("div[class\$=p-b-5]:contains(Pages)").text())
+        }
+        status = SManga.UNKNOWN
+        author = document.select("div[class*=p-t-5]:contains(Author) > div").eachText().joinToString()
+        genre = document.select("div[class*=p-t-5]:contains(Tags) > div:not(:contains(add))").eachText().joinToString()
+    }
+
+    private val dateFormat = SimpleDateFormat("yyyy-MM-dd", Locale.ENGLISH)
+
+    override fun chapterListParse(response: Response): List<SChapter> {
+        val doc = response.asJsoup()
+        val episode = doc.selectFirst(".episode")
+        if (episode != null) {
+            return episode.select("ul > a").reversed().mapIndexed(::chapterFromElement)
+        }
+
+        return listOf(

```

Comment:
```suggestion
        return doc.selectFirst(".episode")?.let {
            episode.select("ul > a").reversed().mapIndexed(::chapterFromElement)
        } ?: listOf(
```

(not tested, might be stupid)